src/server/render/tapestry-render.js

```diff
@@ -0,0 +1,121 @@
+import chalk from 'chalk'
+import isEmpty from 'lodash.isempty'
+
+import prepareAppRoutes from '../routing/prepare-app-routes'
+import matchRoutes from '../routing/match-routes'
+
+import normalizeApiResponse from '../data-fetching/normalize-api-response'
+import fetchFromEndpointConfig from '../data-fetching/fetch-from-endpoint-config'
+import buildErrorView from './error-view'
+import renderSuccessTree from './render-success-tree'
+import renderErrorTree from './render-error-tree'
+
+import baseUrlResolver from '../utilities/base-url-resolver'
+import normaliseUrlPath from '../utilities/normalise-url-path'
+import { log } from '../utilities/logger'
+
+export default async (requestPath, requestQuery, config) => {
```

Comment:
There's an argument that we don't even need to "pass" the `config` (tapestry.config.js) object in here, we could just `import config from tapestry.config.js` as it's aliased in the webpack config.

Reply:
But it feels kinda nice:

`request path + request query + object of route configurations and react components = HTML and a Status code`